### PR TITLE
cuba 4.2 (readd formula)

### PR DESCRIPTION
**Pull request by gipert**
Not migrated formula? See [traces in homebrew-science](https://github.com/Homebrew/homebrew-science/pulls?utf8=✓&q=is%3Apr+cuba).

Cuba is a library for performing multidimensional numerical integrations: <http://www.feynarts.de/cuba/>

PS: this is my first attempt in contributing!

### CODE REVIEW

File: Formula/cuba.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+class Cuba < Formula
+  desc "Library for multidimensional numerical integration"
+  homepage "http://www.feynarts.de/cuba/"
+  url "http://www.feynarts.de/cuba/Cuba-4.2.tar.gz"
+  sha256 "6b75bb8146ae6fb7da8ebb72ab7502ecd73920efc3ff77a69a656db9530a5eef"
+
+  option "without-test", "Skip build-time tests (not recommended)"
```
Comment (fxcoudert) [inline Formula/cuba.rb]: Please remove tests. We only run tests for security software.

File: Formula/cuba.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+  def install
+    ENV.deparallelize
```
Comment (fxcoudert) [inline Formula/cuba.rb]: Does parallel make fail? Has this bug been reported?

File: Formula/cuba.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+    system "./configure", "--prefix=#{prefix}", "--datadir=#{pkgshare}/doc"
```
Comment (fxcoudert) [inline Formula/cuba.rb]: Why is `data` placed in `doc` subdirectory? What would be the default directory?

File: Formula/cuba.rb [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    system "make"
+    system "make", "check" if build.with? "test"
```
Comment (fxcoudert) [inline Formula/cuba.rb]: Remove `make check`

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/cuba.rb (hunk 3 above)
Comment (gipert) [reply Formula/cuba.rb]: I just verified that `"--datadir=#{pkgshare}/doc"` is redundant, I'll remove that

File: Formula/cuba.rb (hunk 2 above)
Comment (gipert) [reply Formula/cuba.rb]: Yes it fails. Frankly I don't know where the development is done, there's [something on Github](https://github.com/JohannesBuchner/cuba) but it doesn't seem to be official. Also, I don't think it's maintained anymore.

Comment (fxcoudert) [reply Formula/cuba.rb]: Please report it to the author (http://wwwth.mpp.mpg.de/members/hahn/) by email, and say so in your comment

File: Formula/cuba.rb [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+  url "http://www.feynarts.de/cuba/Cuba-4.2.tar.gz"
+  sha256 "6b75bb8146ae6fb7da8ebb72ab7502ecd73920efc3ff77a69a656db9530a5eef"
+
+  def install
+    # The provided makefile does not support parallel build
+    ENV.deparallelize
+    system "./configure", "--prefix=#{prefix}"
+    system "make", "install"
+    pkgshare.install "demo"
+  end
```
Comment (fxcoudert) [inline Formula/cuba.rb]: The formula needs a functional test. Either compile some simple C program and link it with the library, or use an example from the provided source or demos.

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/cuba.rb (hunk 5 above)
Comment (gipert) [reply Formula/cuba.rb]: ? Then I don't understand, you asked to remove the test section:

> Please remove tests. We only run tests for security software.

**Comment by gipert** [comment]
I have a question:
This formula is needed as a dependency for another formula I'm currently working on: <https://github.com/bat/bat>. BAT must be compiled with GNU gcc if one wants to enable internal parallelism with OpenMP (`--enable-parallel` flag), otherwise Clang can be used as well. I want to add an option to the BAT formula for this flag that will cause the software to be compiled with GNU gcc instead of Clang. In this case, obviously, I also have to require Cuba to be compiled with GNU gcc. How should I handle this possibility? Should I add to this formula a `--with-gnu-gcc` option and later make BAT depend on Cuba installed with this flag? What are the Homebrew best practices to handle this kind of situations? Is there some documentation outside the Cookbook?

File: Formula/cuba.rb (hunk 5 above)
Comment (fxcoudert) [reply Formula/cuba.rb]: We don't run the software's own test suite during compilation. We run our own functional test, during installation (including when we install prebuilt binaries), to test that the formula installed well and is functioning.

Look at how other formulas have a `test do` block (e.g., `mpfr.rb`).

**Comment by gipert** [comment]
Why isn’t this merged yet? I thought it was ok after the last changes

**Comment by gipert** [comment]
Thanks, this is exactly what I was looking for.

Personally, from a user experience point of view, I would go for building with Clang by default (thus no OpenMP), this mainly because Cuba is a set of libraries, so it could be often used by other Formulas (compiled with Clang in most of the cases, so also Cuba must be compiled with Clang).

However the parallel Cuba works well, why not putting an `with-openmp` option anyways for users interested in advanced usage? The bottle would in any case be compiled with Clang, that is the default we want.

**Review by fxcoudert — APPROVED** [review]

**Comment by fxcoudert** [comment]
Let's merge as is.

**Comment by fxcoudert** [comment]
Thanks @gipert for the pull request and for your patience!

**Pull request closed without merging**